pkg/controller.v1/tensorflow/pod.go

```diff
@@ -292,19 +108,92 @@ func isDistributed(tfjob *tfv1.TFJob) bool {
 	return distributionCount != 1
 }
 
-func setRestartPolicy(podTemplateSpec *v1.PodTemplateSpec, spec *common.ReplicaSpec) {
-	if spec.RestartPolicy == common.RestartPolicyExitCode {
+func setRestartPolicy(podTemplateSpec *v1.PodTemplateSpec, spec *commonv1.ReplicaSpec) {
+	// This is necessary since restartPolicyExitCode is not supported in v1.PodTemplateSpec
+	if spec.RestartPolicy == commonv1.RestartPolicyExitCode {
 		podTemplateSpec.Spec.RestartPolicy = v1.RestartPolicyNever
 	} else {
 		podTemplateSpec.Spec.RestartPolicy = v1.RestartPolicy(spec.RestartPolicy)
 	}
 }
 
-func (tc *TFController) isNonGangSchedulerSet(tfjob *tfv1.TFJob) bool {
-	for _, spec := range tfjob.Spec.TFReplicaSpecs {
-		if spec.Template.Spec.SchedulerName != "" && spec.Template.Spec.SchedulerName != tc.Config.GangSchedulerName {
-			return true
+// IsWorker0Completed return true if pod of worker0 succeeded and exited with 0
+func (tc *TFController) IsWorker0Completed(tfjob *tfv1.TFJob, replicas map[commonv1.ReplicaType]*commonv1.ReplicaSpec) (bool, error) {
```

Comment:
I can not find reference for this methods. Can you point me to the caller?

Reply:
I use it and `PodRestart` in `UpdateJobStatus`.
Since in th-operator, we sometimes update status based on specific pod status (Chief, Master and worker0). With using the framework  from common library, we cannot use updateStatusSingle with flag `restart`, `worker0completed` to updateStatus.
In common library, we now updateStatus after reconcile all pods. So I have to implement these two function to iterate all pods again for specific  scenario.
(That is why I want to know how does mxnet-operator deal with scheduler succeeded in v1 with using common library)